### PR TITLE
Upgrade Psalm to its latest version

### LINKED ISSUE #4221: Inconsistent parameter names
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | yes/no
| Version     | x.y.z

#### Summary
We have signatures that use different names. I want to fix this, but don't know which names are best

<details>

```

ERROR: ParamNameMismatch - lib/Doctrine/DBAL/Driver/DrizzlePDOMySql/Connection.php:13:27 - Argument 1 of Doctrine\DBAL\Driver\DrizzlePDOMySql\Connection::quote has wrong name $value, expecting $input as defined by Doctrine\DBAL\Driver\Connection::quote (see https://psalm.dev/230)
    public function quote($value, $type = ParameterType::STRING)



ERROR: ParamNameMismatch - lib/Doctrine/DBAL/Driver/OCI8/OCI8Connection.php:129:27 - Argument 1 of Doctrine\DBAL\Driver\OCI8\OCI8Connection::quote has wrong name $value, expecting $input as defined by Doctrine\DBAL\Driver\Connection::quote (see https://psalm.dev/230)
    public function quote($value, $type = ParameterType::STRING)


ERROR: ParamNameMismatch - lib/Doctrine/DBAL/Driver/SQLSrv/SQLSrvConnection.php:101:27 - Argument 1 of Doctrine\DBAL\Driver\SQLSrv\SQLSrvConnection::quote has wrong name $value, expecting $input as defined by Doctrine\DBAL\Driver\Connection::quote (see https://psalm.dev/230)
    public function quote($value, $type = ParameterType::STRING)


ERROR: ParamNameMismatch - lib/Doctrine/DBAL/Platforms/OraclePlatform.php:55:52 - Argument 2 of Doctrine\DBAL\Platforms\OraclePlatform::getSubstringExpression has wrong name $position, expecting $from as defined by Doctrine\DBAL\Platforms\AbstractPlatform::getSubstringExpression (see https://psalm.dev/230)
    public function getSubstringExpression($value, $position, $length = null)


ERROR: ParamNameMismatch - lib/Doctrine/DBAL/Platforms/OraclePlatform.php:387:43 - Argument 1 of Doctrine\DBAL\Platforms\OraclePlatform::_getCreateTableSQL has wrong name $table, expecting $tableName as defined by Doctrine\DBAL\Platforms\AbstractPlatform::_getCreateTableSQL (see https://psalm.dev/230)
    protected function _getCreateTableSQL($table, array $columns, array $options = [])


ERROR: ParamNameMismatch - lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php:1108:47 - Argument 1 of Doctrine\DBAL\Platforms\PostgreSqlPlatform::getEmptyIdentityInsertSQL has wrong name $quotedTableName, expecting $tableName as defined by Doctrine\DBAL\Platforms\AbstractPlatform::getEmptyIdentityInsertSQL (see https://psalm.dev/230)
    public function getEmptyIdentityInsertSQL($quotedTableName, $quotedIdentifierColumnName)


ERROR: ParamNameMismatch - lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php:1108:65 - Argument 2 of Doctrine\DBAL\Platforms\PostgreSqlPlatform::getEmptyIdentityInsertSQL has wrong name $quotedIdentifierColumnName, expecting $identifierColumnName as defined by Doctrine\DBAL\Platforms\AbstractPlatform::getEmptyIdentityInsertSQL (see https://psalm.dev/230)
    public function getEmptyIdentityInsertSQL($quotedTableName, $quotedIdentifierColumnName)


ERROR: ParamNameMismatch - lib/Doctrine/DBAL/Platforms/SQLAnywherePlatform.php:1127:45 - Argument 2 of Doctrine\DBAL\Platforms\SQLAnywherePlatform::getTrimExpression has wrong name $pos, expecting $mode as defined by Doctrine\DBAL\Platforms\AbstractPlatform::getTrimExpression (see https://psalm.dev/230)
    public function getTrimExpression($str, $pos = TrimMode::UNSPECIFIED, $char = false)


ERROR: ParamNameMismatch - lib/Doctrine/DBAL/Platforms/SQLServerPlatform.php:895:47 - Argument 1 of Doctrine\DBAL\Platforms\SQLServerPlatform::getEmptyIdentityInsertSQL has wrong name $quotedTableName, expecting $tableName as defined by Doctrine\DBAL\Platforms\AbstractPlatform::getEmptyIdentityInsertSQL (see https://psalm.dev/230)
    public function getEmptyIdentityInsertSQL($quotedTableName, $quotedIdentifierColumnName)


ERROR: ParamNameMismatch - lib/Doctrine/DBAL/Platforms/SQLServerPlatform.php:895:65 - Argument 2 of Doctrine\DBAL\Platforms\SQLServerPlatform::getEmptyIdentityInsertSQL has wrong name $quotedIdentifierColumnName, expecting $identifierColumnName as defined by Doctrine\DBAL\Platforms\AbstractPlatform::getEmptyIdentityInsertSQL (see https://psalm.dev/230)
    public function getEmptyIdentityInsertSQL($quotedTableName, $quotedIdentifierColumnName)


ERROR: ParamNameMismatch - lib/Doctrine/DBAL/Platforms/SQLServerPlatform.php:1071:45 - Argument 2 of Doctrine\DBAL\Platforms\SQLServerPlatform::getTrimExpression has wrong name $pos, expecting $mode as defined by Doctrine\DBAL\Platforms\AbstractPlatform::getTrimExpression (see https://psalm.dev/230)
    public function getTrimExpression($str, $pos = TrimMode::UNSPECIFIED, $char = false)


ERROR: ParamNameMismatch - lib/Doctrine/DBAL/Platforms/SqlitePlatform.php:81:45 - Argument 2 of Doctrine\DBAL\Platforms\SqlitePlatform::getTrimExpression has wrong name $pos, expecting $mode as defined by Doctrine\DBAL\Platforms\AbstractPlatform::getTrimExpression (see https://psalm.dev/230)
    public function getTrimExpression($str, $pos = TrimMode::UNSPECIFIED, $char = false)


ERROR: ParamNameMismatch - lib/Doctrine/DBAL/Platforms/SqlitePlatform.php:106:52 - Argument 2 of Doctrine\DBAL\Platforms\SqlitePlatform::getSubstringExpression has wrong name $position, expecting $from as defined by Doctrine\DBAL\Platforms\AbstractPlatform::getSubstringExpression (see https://psalm.dev/230)
    public function getSubstringExpression($value, $position, $length = null)


ERROR: ParamNameMismatch - lib/Doctrine/DBAL/Platforms/SqlitePlatform.php:327:43 - Argument 1 of Doctrine\DBAL\Platforms\SqlitePlatform::_getCreateTableSQL has wrong name $name, expecting $tableName as defined by Doctrine\DBAL\Platforms\AbstractPlatform::_getCreateTableSQL (see https://psalm.dev/230)
    protected function _getCreateTableSQL($name, array $columns, array $options = [])


ERROR: ParamNameMismatch - lib/Doctrine/DBAL/Portability/Statement.php:116:46 - Argument 2 of Doctrine\DBAL\Portability\Statement::setFetchMode has wrong name $arg1, expecting $arg2 as defined by Doctrine\DBAL\Driver\ResultStatement::setFetchMode (see https://psalm.dev/230)
    public function setFetchMode($fetchMode, $arg1 = null, $arg2 = null)


ERROR: ParamNameMismatch - lib/Doctrine/DBAL/Portability/Statement.php:116:60 - Argument 3 of Doctrine\DBAL\Portability\Statement::setFetchMode has wrong name $arg2, expecting $arg3 as defined by Doctrine\DBAL\Driver\ResultStatement::setFetchMode (see https://psalm.dev/230)
    public function setFetchMode($fetchMode, $arg1 = null, $arg2 = null)

ERROR: ParamNameMismatch - lib/Doctrine/DBAL/Schema/DrizzleSchemaManager.php:94:53 - Argument 1 of Doctrine\DBAL\Schema\DrizzleSchemaManager::_getPortableTableIndexesList has wrong name $tableIndexes, expecting $tableIndexRows as defined by Doctrine\DBAL\Schema\AbstractSchemaManager::_getPortableTableIndexesList (see https://psalm.dev/230)
    protected function _getPortableTableIndexesList($tableIndexes, $tableName = null)


ERROR: ParamNameMismatch - lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php:78:53 - Argument 1 of Doctrine\DBAL\Schema\MySqlSchemaManager::_getPortableTableIndexesList has wrong name $tableIndexes, expecting $tableIndexRows as defined by Doctrine\DBAL\Schema\AbstractSchemaManager::_getPortableTableIndexesList (see https://psalm.dev/230)
    protected function _getPortableTableIndexesList($tableIndexes, $tableName = null)


ERROR: ParamNameMismatch - lib/Doctrine/DBAL/Schema/OracleSchemaManager.php:95:53 - Argument 1 of Doctrine\DBAL\Schema\OracleSchemaManager::_getPortableTableIndexesList has wrong name $tableIndexes, expecting $tableIndexRows as defined by Doctrine\DBAL\Schema\AbstractSchemaManager::_getPortableTableIndexesList (see https://psalm.dev/230)
    protected function _getPortableTableIndexesList($tableIndexes, $tableName = null)


ERROR: ParamNameMismatch - lib/Doctrine/DBAL/Schema/PostgreSqlSchemaManager.php:217:53 - Argument 1 of Doctrine\DBAL\Schema\PostgreSqlSchemaManager::_getPortableTableIndexesList has wrong name $tableIndexes, expecting $tableIndexRows as defined by Doctrine\DBAL\Schema\AbstractSchemaManager::_getPortableTableIndexesList (see https://psalm.dev/230)
    protected function _getPortableTableIndexesList($tableIndexes, $tableName = null)


ERROR: ParamNameMismatch - lib/Doctrine/DBAL/Schema/SqliteSchemaManager.php:175:53 - Argument 1 of Doctrine\DBAL\Schema\SqliteSchemaManager::_getPortableTableIndexesList has wrong name $tableIndexes, expecting $tableIndexRows as defined by Doctrine\DBAL\Schema\AbstractSchemaManager::_getPortableTableIndexesList (see https://psalm.dev/230)
    protected function _getPortableTableIndexesList($tableIndexes, $tableName = null)


```

</details>

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

File: lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php [hunk 1]
```diff
@@ -838,15 +838,15 @@ protected function _getPortableTableColumnList($table, $database, $tableColumns)
     /**
      * Aggregates and groups the index results according to the required data result.
      *
-     * @param mixed[][]   $tableIndexRows
+     * @param mixed[][]   $tableIndexes
```
Comment (greg0ire) [inline lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php:841]: Which one do you prefer @morozov ?

Comment (morozov) [reply lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php:841]: `$tableIndexes` is okay.

**Comment by greg0ire** [comment]
> The inferred type 'int' does not match the declared return type 'string' for Doctrine\DBAL\Driver\Mysqli\MysqliConnection::lastInsertId

What should be done about that error? We can either widen the type in interface, but the description says it is supposed to be a `string` representation… should that error be suppressed, and addressed in the next major by casting to string?

**Comment by morozov** [comment]
> What should be done about that error?

Let's suppress for now. This API was attempted to be fixed for a couple of years now and it's really questionable per se.

File: lib/Doctrine/DBAL/Driver/DrizzlePDOMySql/Connection.php [hunk 2]
```diff
@@ -10,12 +10,12 @@ class Connection extends PDOConnection
     /**
      * {@inheritdoc}
      */
-    public function quote($value, $type = ParameterType::STRING)
```
Comment (morozov) [inline lib/Doctrine/DBAL/Driver/DrizzlePDOMySql/Connection.php:13]: Let's use `$value` everywhere. The fact that only input needs to be quoted is $1M misconception.

File: lib/Doctrine/DBAL/Platforms/OraclePlatform.php [hunk 3]
```diff
@@ -52,13 +52,13 @@ public static function assertValidIdentifier($identifier)
     /**
      * {@inheritDoc}
      */
-    public function getSubstringExpression($value, $position, $length = null)
+    public function getSubstringExpression($value, $from, $length = null)
```
Comment (morozov) [inline lib/Doctrine/DBAL/Platforms/OraclePlatform.php:55]: Please make sure this is consistent with `master` (https://github.com/doctrine/dbal/pull/3494). The idea is to have the arguments match ANSI SQL analogs if possible (e.g.[ `SUBSTR()`](https://www.w3schools.com/sql/func_mysql_substr.asp)).

File: lib/Doctrine/DBAL/Platforms/OraclePlatform.php [hunk 4]
```diff
@@ -384,11 +384,11 @@ public function getListSequencesSQL($database)
     /**
      * {@inheritDoc}
      */
-    protected function _getCreateTableSQL($table, array $columns, array $options = [])
+    protected function _getCreateTableSQL($tableName, array $columns, array $options = [])
```
Comment (morozov) [inline lib/Doctrine/DBAL/Platforms/OraclePlatform.php:387]: This is against the approach taken in https://github.com/doctrine/dbal/pull/4172:
> Argument names have been simplified where possible (e.g. `dropTable($tableName)` → `dropTable($name)`).

Unless we're dealing both with a table and a name in the same context, there's no much point in being too specific IMO.

File: lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php [hunk 5]
```diff
@@ -1105,9 +1105,9 @@ public function getDateTimeTzFormatString()
     /**
      * {@inheritDoc}
      */
-    public function getEmptyIdentityInsertSQL($quotedTableName, $quotedIdentifierColumnName)
```
Comment (morozov) [inline lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php:1108]: The "quoted" attribute should stay as is. It's a rabbit hole but it has its meaning. I can provide more details if necessary.

Comment (greg0ire) [reply lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php:1108]: I trust you :)

File: lib/Doctrine/DBAL/Platforms/AbstractPlatform.php [hunk 6]
```diff
@@ -1696,13 +1696,13 @@ public function getInlineColumnCommentSQL($comment)
...
      *
-     * @param string    $tableName
+     * @param string    $name
      * @param mixed[][] $columns
      * @param mixed[]   $options
      *
      * @return string[]
      */
-    protected function _getCreateTableSQL($tableName, array $columns, array $options = [])
+    protected function _getCreateTableSQL($name, array $columns, array $options = [])
```
Comment (greg0ire) [inline lib/Doctrine/DBAL/Platforms/AbstractPlatform.php:1705]: Ah this must a reason why the tests fail… `$name` is already taken

**Review by morozov — COMMENTED** [review]
Should the Psalm version constraint be updated as well? Otherwise, looks good.

**Comment by greg0ire** [comment]
Sure, let's prevent accidental downgrades :+1:

**Review by morozov — APPROVED** [review]

**Pull request merged**